src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_instruct_pix2pix.py

```diff
@@ -491,15 +543,19 @@ def _encode_prompt(self, prompt, device, num_images_per_prompt, do_classifier_fr
             )
             uncond_embeddings = uncond_embeddings[0]
 
+        if do_classifier_free_guidance:
             # duplicate unconditional embeddings for each generation per prompt, using mps friendly method
             seq_len = uncond_embeddings.shape[1]
+
+            uncond_embeddings = uncond_embeddings.to(dtype=self.text_encoder.dtype, device=device)
+
             uncond_embeddings = uncond_embeddings.repeat(1, num_images_per_prompt, 1)
             uncond_embeddings = uncond_embeddings.view(batch_size * num_images_per_prompt, seq_len, -1)
 
             # For classifier free guidance, we need to do two forward passes.
             # Here we concatenate the unconditional and text embeddings into a single batch
             # to avoid doing two forward passes
-            text_embeddings = torch.cat([text_embeddings, uncond_embeddings, uncond_embeddings])
+            text_embeddings = torch.cat([uncond_embeddings, text_embeddings])
```

Comment:
I think this is a typo? InstructPix2Pix requires 3 sets of latents, and the order is different in this pipeline for some reason (it's text + uncond + uncond whereas in the others it's uncond + text).

Reply:
Good catch! And yes, the order follows the original implem.

Reply:
It's because I added the Copy-from statement, I order them correctly after the function :-)

Reply:
Oh, I see. Yeah, I missed that. I'm hesitant about the convenience of adding the copied-from comment vs the potential loss in legibility. I suggested a more verbose comment above.

Reply:
Yeah it's not ideal, I'll get the "# End Copy statement working, so then it should be much more readable